internal/pkgecosystem/package.go

```diff
@@ -1,42 +1,50 @@
 package pkgecosystem
 
-type Pkg struct {
-	name    string
-	version string
-	manager *PkgManager
-	local   string
-	command string
+import "fmt"
+
+// Package provides an interface for working with packages from any supported package ecosystem
+type Package interface {
+	// Name returns the name of this package
+	Name() string
+	// Ecosystem returns the ecosystem of this package. It is returned as a string for compatibility reasons,
+	// but is always a valid Ecosystem.
+	Ecosystem() string
+	// Version returns the version of this package
+	Version() string
+	// LocalPath specifies a path to a local archive associated with this package, if one exists.
+	LocalPath() string
+	// Command returns the full command (with arguments) for dynamic analysis
+	Command(phase RunPhase) []string
+	// Download attempts to download an archive for the given package name and version and returns the archive path
+	Download() (string, error)
+	// DynamicAnalysisImage returns the name of container image to use for dynamic analysis
+	DynamicAnalysisImage() string
```

Comment:
Nit: It seems like these ones (DynamicAnalysisImage, DynamicRunPhases) deserve to be part of a different interface. It's good to keep interfaces more minimal and focused.

Reply:
Yeah good point. I can make these into a separate interface, since they really only depend on the ecosystem